File: Java/Bookyard.Contracts/src/practice/bookyard/contracts/beans/BookRecommendations.java
package practice.bookyard.contracts.beans;

import java.util.ArrayList;
import java.util.List;

public class BookRecommendations {
	private int userId;
	private String userName;
	private List<Book> books;
	
	public BookRecommendations() {
		this(0, null);
	}
	
	public BookRecommendations(int userId, String userName) {
		this.setUserId(userId);
		this.setUserName(userName);
		
		this.books = new ArrayList<Book>();
	}
	
	public int getUserId() {
		return userId;
	}
	public void setUserId(int userId) {
		this.userId = userId;
	}
	public String getUserName() {
		return userName;
	}
	public void setUserName(String userName) {
		this.userName = userName;
	}
	
	public List<Book> getBooks() {
		return this.books;
	}
	
	public void addBook(Book book) {
		if (book == null) return;
		
		if (this.books.contains(book)) return;
		
		this.books.add(book);
	}
	
	public void removeBook(Book book) {
		if (book == null) return;
		
		if (this.books.contains(book)) {
			this.books.remove(book);
		}
	}
}

File: Contracts/src/BookRecommendations.kt
package bookyard.contracts.beans;

import java.util.ArrayList;

data class BookRecommendations(var userId : Int = 0,
                               var userName : String? = null,
                               val books : MutableList<Book?> = ArrayList<Book?>()) {

    fun addBook(book : Book?) {
        if (book == null) return;

        if (this.books.contains(book)) return;

        this.books.add(book);
    }

    fun removeBook(book : Book?) {
        if (book == null) return;

        if (this.books.contains(book)) {
            this.books.remove(book);
        }
    }
}